python/mp/rosary.py

```diff
@@ -148,6 +156,54 @@ def register_defined_effects(self):
                 self.register_effect(eff)
 
 
+    def register_trigger(self, trigger_class):
+
+        print("REGISTER A TRIGGER")
+        print(trigger_class)
+
+        # Instantiate
+        self.trigger_id += 1
+        trigger = trigger_class()
+        trigger.rosary = self
+        trigger.id = self.trigger_id
+        self.triggers[trigger.name] = trigger
+
+        print("WHAT ARE MY TRIGGERS")
+        print(self.triggers)
+
+        self.dispatcher.map(trigger.osc_path,
+                            trigger.trigger_wrapper,
+                            trigger)
+
+
+    def find_defined_triggers(self, module_or_class):
+        """
+        Crawl through mp.triggers package and find (non-abstract) subclasses
+        to register with the rosary
+        """
+
+        classes = set()
+        for name, obj in inspect.getmembers(module_or_class):
+            if inspect.ismodule(obj) and obj.__package__ == 'mp.triggers':
+                classes = classes.union(self.find_defined_triggers(obj))
+            elif inspect.isclass(obj):
+                classes.add(obj)
+
+        return classes
+        
+
+    def register_defined_triggers(self):
+        """
+        Figure out which triggers are defined in .py files, and register them
+        """
+
+        defined_triggers = self.find_defined_triggers(triggers)
+        for tr in defined_triggers:
+            # Don't register abstract classes, e.g. effects.effect.Effect
+            if not inspect.isabstract(tr) and issubclass(tr, triggers.trigger.Trigger):
+                self.register_trigger(tr)
```

Comment:
A lot of copy/paste I realize. But for the sake of the demo I won't lose sleep over this.